traits/observers/expression.py

```diff
@@ -76,11 +85,82 @@ def then(self, expression):
         """
         return SeriesExpression(self, expression)
 
+    def dict_items(self, notify=True, optional=False):
+        """ Create a new expression for observing items inside a dict.
+
+        Events emitted (if any) will be instances of
+        :py:class:`~traits.observers.events.DictChangeEvent`.
+
+        If an expression with ``dict_items`` is further extended, the
+        **values** of the dict will be given to the next item in the
+        expression. For example, the following observes a trait named "number"
+        on any object that is one of the values in the dict named "mapping"::
+
+            trait("mapping").dict_items().trait("number")
+
+        Parameters
+        ----------
+        notify : boolean, optional
+            Whether to notify for changes.
```

Comment:
Please mention the default in the docstring (also applies to `optional` and to similar methods below).